src/components/PendingTxsList/index.tsx

```diff
@@ -0,0 +1,191 @@
+  const [queuedTransactions, setQueuedTransactions] = useState<TransactionSummary[]>([])
+
+  useEffect(() => {
+    if (isLoadingOwnerSafes === undefined) return
+
+    if (ownerSafes[chainId]?.length) {
+      setSelectedSafe(ownerSafes[chainId][0])
+    } else {
+      setSelectedSafe(defaultSafe)
+    }
+    setLoadingOwnerSafes(false)
+  }, [chainId, defaultSafe, isLoadingOwnerSafes, ownerSafes, selectedSafe])
+
+  // Fetch txs awaiting confirmations
+  useEffect(() => {
+    if (selectedSafe === undefined) return
+
+    let isCurrent = true
+    const fetchQueuedTxs = async () => {
+      const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(selectedSafe))
+      const txs = results
+        .filter((result) => isTransactionSummary(result))
+        .slice(0, MAX_TXS_DISPLAY)
+        .map((item: TransactionListItem) => (item as Transaction).transaction)
+      if (isCurrent) {
+        setQueuedTransactions(txs)
+        setIsFetchingPendingTxs(false)
+      }
+    }
+    setIsFetchingPendingTxs(true)
+    fetchQueuedTxs()
+
+    return () => {
+      isCurrent = false
+    }
+  }, [chainId, selectedSafe])
+
+  if (isFetchingPendingTxs) {
+    return (
+      <>
+        {Array.from(Array(MAX_TXS_DISPLAY).keys()).map((_, idx) => (
+          <TransactionSkeleton key={idx} />
+        ))}
+      </>
+    )
+  }
+
+  if (selectedSafe && !queuedTransactions.length && isFetchingPendingTxs === false) {
+    return <h3>This Safe has no pending transactions</h3>
+  }
+
+  if (!loadingOwnerSafes && !selectedSafe) {
+    return <h3>No Safes to display</h3>
+  }
+
+  if (!selectedSafe) return <></>
+
+  const { shortName } = getChainById(chainId)
+  const url = generateSafeRoute(SAFE_ROUTES.TRANSACTIONS_QUEUE, { safeAddress: selectedSafe, shortName })
+  return (
+    <List component="div">
+      {queuedTransactions.map((transaction) => (
+        <PendingTx transaction={transaction} url={url} key={transaction.id} />
+      ))}
+    </List>
+  )
+}
+
+type PendingTxType = {
+  transaction: TransactionSummary
+  url: string
+}
+const PendingTx = ({ transaction, url }: PendingTxType) => {
+  const info = useAssetInfo(transaction.txInfo)
+  const type = useTransactionType(transaction)
+  const toAddress = getTxTo(transaction)
+  const toInfo = useKnownAddress(toAddress)
+
+  return (
+    <TransactionToConfirm key={transaction.id} to={url}>
+      <Text color="text" size="lg" as="span">
+        {isMultisigExecutionInfo(transaction.executionInfo) && transaction.executionInfo.nonce}
+      </Text>
+      <CustomIconText
+        address={toAddress?.value || '0x'}
+        iconUrl={type.icon || toInfo?.logoUri || undefined}
+        iconUrlFallback={type.fallbackIcon}
+        text={type.text || toInfo?.name || undefined}
+      />
+      {info ? <TxInfo info={info} /> : <Spacer />}
+      <TxConfirmations>
+        {isMultisigExecutionInfo(transaction.executionInfo) ? (
```

Comment:
If you move the `isMultisigExecutionInfo(transaction.executionInfo)` check further up and return `null` if not true, you won't need to check this twice.

Reply:
That would apply for the conditional rendering in the first occurrence but not for the ternary in the second.

Reply:
Aren't *all* multisig transactions?